spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -136,6 +141,8 @@ data class SpdxCompoundExpression(
     val operator: SpdxOperator,
     val right: SpdxExpression
 ) : SpdxExpression() {
+    override fun copy() = copy(left = left, operator = operator, right = right)
```

Comment:
For a deep copy this should be `copy(left = left.copy(), operator = operator, right = right.copy())`.

Reply:
The last implementation I had was a mix of your and mine proposals, and added this to the sealed base class (no other code changes necessary):

    fun copy(): SpdxExpression =
        when (this) {
            is SpdxCompoundExpression -> copy(left = left.copy(), operator = operator, right = right.copy())
            is SpdxLicenseExceptionExpression -> copy(id = id)
            is SpdxLicenseIdExpression -> copy(id = id, orLaterVersion = orLaterVersion)
            is SpdxLicenseReferenceExpression -> copy(id = id)
        }

But even that is not needed anymore, see my comment when I closed the PR.